app/controllers/application_controller.rb

```diff
@@ -1,7 +1,7 @@
 class ApplicationController < ActionController::Base
   protect_from_forgery with: :exception
 
-  helper_method :cart, :current_user
+  helper_method :cart, :current_user, :current_visitor?
```

Comment:
Awesome update to also check current_visitor? (a method I believe you made?)

Reply:
This was added to prevent a user from being able to view a profile if just a visitor. It's being called in #show in `users_contoller` but maybe there is a better way to do this validation? I like how in other controllers, we prohibit access with the `base_controller` but I'm not sure if there's a way to block a user off from just certain controller actions and not others?

Reply:
Hmmm.... I don't know either. We can take this to our mentors or ask during our checkin